update env vars for document handling (#5367)
camunda/camunda-docs: docs/components/concepts/document-handling.md

:::

| Store variable             | Required | Description                                                                                                                                 |
| -------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
| DOCUMENT_STORE_LOCAL_PATH  | Yes      | The path to the directory which will host the uploaded files.                                                                               |
| DOCUMENT_STORE_LOCAL_CLASS | Yes      | The class for instantiating the local store. This must always be `io.camunda.document.store.localstorage.LocalStorageDocumentStoreProvider` |

**Example:**

```
DOCUMENT_STORE_LOCAL_CLASS=io.camunda.document.store.localstorage.LocalStorageDocumentStoreProvider
DOCUMENT_STORE_LOCAL_PATH=/usr/local/camunda
DOCUMENT_DEFAULT_STORE_ID=local
```

</TabItem>

<TabItem value='in-memory'>

:::note
In-memory storage is not usable in production.
:::

If no configuration is provided and no `DOCUMENT_DEFAULT_STORE_ID` is set, in-memory is used as the default storage type.

> If configurations are provided for at least one storage type (DOCUMENT_STORE_AWS_BUCKET, DOCUMENT_STORE_AWS_BUCKET_PATH, etc.) ensure `DOCUMENT_DEFAULT_STORE_ID=in-memory` is set for in-memory storage.

Reviewer: This reads a little ambiguously; I assume it means the following:

```suggestion
If configurations are provided for at least one storage type (DOCUMENT_STORE_AWS_BUCKET, DOCUMENT_STORE_AWS_BUCKET_PATH, etc.) `DOCUMENT_DEFAULT_STORE_ID=in-memory` must be set to enable in-memory storage.
```

I have the same question regarding `in-memory` here as below.

> | Store variable                | Required | Description                                                                                                                                                                                                                               |
> | ----------------------------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
> | DOCUMENT_STORE_INMEMORY_CLASS | Yes      | The class for instantiating the in-memory store. This must always be `io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider`. If this is set, Be sure to remove the hyphen and set `DOCUMENT_DEFAULT_STORE_ID` to `=inmemory`. |

Reviewer: The in-memory storage section is hard for me to follow - line 146 says to set `in-memory`, but this table asks to remove the hyphen? Is there a case in which it would be set to `in-memory` and not `inmemory`? 

```suggestion
| Store variable                | Required | Description                                                                                                                                                                                                                               |
| ----------------------------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| DOCUMENT_STORE_INMEMORY_CLASS | Yes      | The class for instantiating the in-memory store. This must always be `io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider`. If this is set, also set `DOCUMENT_DEFAULT_STORE_ID` to `inmemory`. |
```

I've given a suggestion here, but I think this would be easier to follow if the `DOCUMENT_DEFAULT_STORE_ID` was defined in each tab's table. This would allow you to remove line 32 ("To set what storage should be used..."), which is easily missed when scanning the tables.

Reviewer 2: Thank you for bringing this up @conceptualshark. I don't think what is written on line 146 about the use of `DOCUMENT_DEFAULT_STORE_ID=in-memory` is correct. 

Users can specify the env vars for multiple stores (AWS, GCP, local) if they want, but only one store can be used at a time. If users want to use the in-memory store and:
* they have explciitly specified env vars for other stores, then they need to set `DOCUMENT_STORE_INMEMORY_CLASS=io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider` and `DOCUMENT_DEFAULT_STORE_ID=inmemory`
* they have not explicitly specified env vars for any store, then our application will use in-memory by default

Does this clarify things?

Author: Adjusting this in an incoming commit -- let me know what you think.

| `DOCUMENT_STORE_LOCAL_CLASS` | Yes      | The class for instantiating the local store. This must always be `io.camunda.document.store.localstorage.LocalStorageDocumentStoreProvider` |

**Example:**

```
DOCUMENT_STORE_LOCAL_CLASS=io.camunda.document.store.localstorage.LocalStorageDocumentStoreProvider
DOCUMENT_STORE_LOCAL_PATH=/usr/local/camunda
DOCUMENT_DEFAULT_STORE_ID=local
```

</TabItem>

<TabItem value='in-memory'>

:::note
In-memory storage is not usable in production.
:::

You can specify the environment variables for multiple stores (`DOCUMENT_STORE_AWS_BUCKET`, `DOCUMENT_STORE_AWS_BUCKET_PATH`, etc.) but only one store can be used at a time.

To use the in-memory store, take the following steps:

1. Explicitly specify the variables for the other stores. If you do not explicitly specify variables for any store, our application will use in-memory by default.
2. Set `DOCUMENT_STORE_INMEMORY_CLASS=io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider`.
3. Set `DOCUMENT_DEFAULT_STORE_ID=inmemory`.

> If no configuration is provided for at least one storage type, and no `DOCUMENT_DEFAULT_STORE_ID` is set, in-memory is used as the default storage type.

Reviewer: I think this might be able to be reordered a bit to clarify - in-memory storage is the default, but if you apply another config and want to use in-memory storage, you _must_ set these variables. So setting `inmemory` will override any other config, but isn't needed if no alternate config exists.

--

If no configuration is provided for at least one storage type, and no `DOCUMENT_DEFAULT_STORE_ID` is set, in-memory is used as the default storage type. If the configuration for another storage type has been provided (`DOCUMENT_STORE_AWS_BUCKET`, `DOCUMENT_STORE_AWS_BUCKET_PATH`, etc.), in-memory storage must be set explicitly in order to be used.

To use the in-memory store when an alternate configuration has been provided, take the following steps:

1. Set `DOCUMENT_STORE_INMEMORY_CLASS=io.camunda.document.store.inmemory.InMemoryDocumentStoreProvider`.
2. Set `DOCUMENT_DEFAULT_STORE_ID=inmemory`.

Author: Thank you! I have updated the PR.
